pkg/apis/pipeline/v1beta1/param_types_test.go

```diff
@@ -282,6 +286,48 @@ func TestArrayOrString_UnmarshalJSON(t *testing.T) {
 	}
 }
 
+func TestArrayOrString_UnmarshalJSON_Directly(t *testing.T) {
+	cases := []struct {
+		desc     string
+		input    string
+		expected v1beta1.ArrayOrString
+	}{
+		{desc: "empty value", input: ``, expected: *v1beta1.NewArrayOrString("")},
+		{desc: "int value", input: `1`, expected: *v1beta1.NewArrayOrString("1")},
+		{desc: "int array", input: `[1,2,3]`, expected: *v1beta1.NewArrayOrString("[1,2,3]")},
+		{desc: "nested array", input: `[1,\"2\",3]`, expected: *v1beta1.NewArrayOrString(`[1,\"2\",3]`)},
+		{desc: "string value", input: `hello`, expected: *v1beta1.NewArrayOrString("hello")},
+		{desc: "array value", input: `["hello","world"]`, expected: *v1beta1.NewArrayOrString("hello", "world")},
+	}
```

Comment:
thank you @Yongxuanzhang for adding a list of unit tests 🙏 

NIT: please add tests for unmarshalling map as well, not a blocker though.

Reply:
Sure! Np!